components/form.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useState, ChangeEvent, FormEvent, FC } from 'react';
+import styled from 'styled-components';
+import Heading1 from '../elements/Heading1';
+import Heading2 from '../elements/Heading3';
+import Heading3 from '../elements/Heading3';
+import Paragraph1 from '../elements/Paragraph1';
+import SubmitButton from '../elements/SubmitButton';
+import FormElement from '../elements/FormElement';
+import TextInput from '../elements/TextInput';
+import { postTestimonial } from '../services/airTableService';
+import testimonial from '../interfaces/testimonial';
+
+// const FormElement = styled.form`
+//   margin: 2rem 2rem;
+//   width: 80%;
+//   display: flex;
+//   flex-direction: column;
+//   justify-content: center;
+//   align-items: center;
+//   box-sizing: border-box;
+//   box-shadow: rgba(50, 50, 93, 0.25) 0px 50px 100px -20px,
+//     rgba(0, 0, 0, 0.3) 0px 30px 60px -30px;
+//   border-radius: 1rem;
+// `;
+
+const Form: FC = () => {
+  const emptyForm: testimonial = {
+    testimony: '',
+    location: '',
+    consent: false,
+  };
+  const [testimonial, setTestimonial] = useState(emptyForm);
+
+  const handleChange = (
+    e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ): void => {
+    const name: string = e.target.name;
+    const value: string | boolean =
+      e.target.type === 'checkbox' ? !testimonial.consent : e.target.value;
+
+    setTestimonial(prevTestimonial => ({
+      ...prevTestimonial,
+      [name]: value,
+    }));
+  };
+
+  const handleSubmit = async (e: FormEvent<HTMLFormElement>): Promise<void> => {
+    e.preventDefault();
+    await postTestimonial(testimonial);
```

Comment:
we need to handle the scenario where there is an error when we try to submit the form

Reply:
we'll use a 
```
try {
await api request
// do something with success

} catch (e) {
// do something with error

}
```